external/llvm-project/mlir/test/Dialect/GPU/async-region.mlir

```diff
@@ -177,9 +177,9 @@ module attributes {gpu.container_module} {
     %t0 = gpu.wait async
     // CHECK: %[[t1:.*]] = gpu.wait async [%[[t0]]]
     %t1 = gpu.wait async [%t0]
-    // CHECK: %[[m:.*]], %[[t2:.*]] = gpu.alloc async [%[[t1]], %[[t0]]] ()
+    // CHECK: %[[m:.*]], %[[t2:.*]] = gpu.alloc async [%[[t0]]] ()
     %0 = gpu.alloc [%t0] () : memref<7xf32>
-    // CHECK: %[[t3:.*]] = gpu.dealloc async [%[[t2]]] %[[m]]
+    // CHECK: %[[t3:.*]] = gpu.dealloc async [%[[t2]], %[[t1]]] %[[m]]
```

Comment:
How's the dependency on t1 end up down here? 

I take it that's because this is the first sync op after `t1` was created? If so, probably a good idea to add a comment.

Reply:
It picks up these dependencies because it has no deps.. 
The new change for partially async'd graph is:
* all ops get a token output except a gpu.wait
* only ops that have no dependencies get the current set of unused tokens

In this example, the dealloc is the first op that doesn't have deps and the alloc token is not used so it is added as well as the extra wait at the beginning.

Reply:
Ah, cool